pkg/controllers/infracluster/aws.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package infracluster
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	cerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	awsv1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ensureAWSCluster ensures the AWSCluster cluster object exists.
+func (r *InfraClusterController) ensureAWSCluster(ctx context.Context, log logr.Logger) (client.Object, error) {
+	target := &awsv1.AWSCluster{ObjectMeta: metav1.ObjectMeta{
+		Name:      r.Infra.Status.InfrastructureName,
+		Namespace: defaultCAPINamespace,
+	}}
+
+	// Checking whether InfraCluster object exists. If it doesn't, create it.
```

Comment:
In what cases would it exist (or not)?

Reply:
Yes I guess it can be created by the (CAPI based) Installer and pivoted, or it can be created via Install provided Manifests.
This does account for those cases and does not override it